reference-lib/src/main/java/org/veriblock/integrations/VeriBlockSecurity.java

```diff
@@ -274,25 +274,33 @@ private void verifyPublicationContextually(AltPublication publication) throws Ve
         checkVeriBlockContextually(publication.getFirstBlock());
     }
 
-    private void checkVeriBlockContextually(VeriBlockBlock firstBlock) throws BlockStoreException, SQLException {
+    void checkVeriBlockContextually(VeriBlockBlock firstBlock) throws BlockStoreException, SQLException {
         if (firstBlock == null) {
             throw new VerificationException("Publication does not have any VeriBlock blocks");
         }
 
         VeriBlockBlock previous = veriblockBlockchain.searchBestChain(firstBlock.getPreviousBlock());
         if (previous == null) {
-            throw new VerificationException("Publication does not connect to VeriBlock blockchain");
+            // corner case: the first bootstrap block has no previous block
+            // but does connect to the blockchain by definition
+            if (veriblockBlockchain.searchBestChain(firstBlock.getHash()) == null) {
```

Comment:
I think the real meaning of this code is to check if the block is a bootstrap block or already existing block. So we could rename this check, eg:
if (!isExistingBlock(firstBlock)) throw

Reply:
Not at all. It's typical for this code to be called with a block that is not in the store but connects to a block that is already in the store. Renamed to checkConnectivity.

The only ugly part is: throw new VerificationException("Publication does not have any VeriBlock blocks");

I'd rather move it out of the function, but then it would cause code duplication :(